File: ICHI_API/Model/ProductDTO.cs
using ICHI_CORE.Domain.MasterModel;

namespace ICHI_API.Model
{
    public class ProductDTO
    {
        public Product Product { get; set; }
        public IEnumerable<ProductImages> ProductImages { get; set; }
        public Category CategoryProduct { get; set; }
        public PromotionDetail PromotionDetail { get; set; }

        // số lượng đã bán
        public int quantitySold { get; set; }
        // số lượng đang trên đơn

        public int QuantityOnnTheOrder { get; set; }
    }
}

File: ICHI_Client/src/app/dtos/product.dto.ts
import { ProductModel } from './../models/product.model';
import { CategoryProduct } from '../models/category.product';
import {
  IsString,
  IsNotEmpty,
  IsPhoneNumber,
  IsBoolean,
  IsNumber,
} from 'class-validator';
import { MasterEntityDTO } from './master.entity.dto';
import { ProductImage } from '../models/product.image';

export class ProductDTO {
  product: ProductModel;
  categoryProduct: CategoryProduct;
  productImages: ProductImage[];
  quantitySold: number;
  quantityOnnTheOrder: number;
  constructor(
    product: ProductModel,
    category: CategoryProduct,
    ProductImages: ProductImage[],
    quantitySold: number,
    quantityOnnTheOrder: number
  ) {
    this.product = product;
    this.categoryProduct = category;
    this.productImages = ProductImages;
    this.quantitySold = quantitySold;
    this.quantityOnnTheOrder = quantityOnnTheOrder;
  }
}
